### PR TITLE
Michael's superhero

### PR DESCRIPTION
**Pull request by mroyt**
i use the timer to refresh stamina and such, and im having trouble
figuring out how to test for that. i also have a function (reduceStamina) that I just can't get to work. it keeps saysing stamina is NaN for some reason

### CODE REVIEW

File: js/super_hero.js [hunk 1]
```diff
@@ -1 +1,83 @@
-// var SuperHero = ...
+
+var SuperHero = function(name,heightInCms,weightInKgs,strength,toughness,speed,stamina,health){
```
Comment (afeld) [inline js/super_hero.js:2]: When you have this many options, you might want to accept them as an `options` object, but have defaults so they don't all need to be passed in.

Comment (mroyt) [inline js/super_hero.js:2]: ah, i'd never heard of the options object. i'll have to look into that.
yea, that would probably be helpful

Comment (afeld) [inline js/super_hero.js:2]: It's not a special thing – just means that you have a function that takes an object as an argument that contains various options.

``` javascript
var myFunc = function(opts){
  this.name = opts.name;
  // ...
};

myFunc({
  name: 'foo',
  // ...
});
```

**Comment by afeld** [comment]
> The hard part was getting the tests to pass with the weird timing stuff I build into it

Yeah, dealing with time is tough.  Two approaches are
- Passing the "current" time to the function, which can be 60+ seconds in the future to pretend that the time has passed
- Faking the clock using a library like Sinon: http://sinonjs.org/docs/#clock

> when you have multiple calls to the same function, things get messed up

True!  For exactly this reason, you'll generally want to have no shared state between the tests.  In other words, create the `SuperHero` within each test, rather than just once.

Nice work!

**Pull request closed without merging**